security/pkg/nodeagent/sds/sdsservice_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package sds
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"testing"
+	"time"
+
+	api "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	authapi "github.com/envoyproxy/go-control-plane/envoy/api/v2/auth"
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	sds "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
+	"github.com/gogo/protobuf/types"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+var (
+	socket               string
```

Comment:
Do you need the socket to be global var? I saw we pass them as function param.

Reply:
yes, it's also used in unixDialer

Reply:
unixDialer is the callback and uses its parameter `socket` not the global variable. I think if you pass `socket` to `setupConnection(socket string)` then we don't need the global var?

My main concern is that `socket` is a too common name and can easily shadowed by local var.

https://github.com/istio/istio/pull/5913/files/5b7aec1bcdd8a97eed61eaaed54ffd4d003a67d0#diff-0a5f4e6aef49f7c782da2534e7d36f99R117

Reply:
done